kvbc/cmf/categorized_kvbc_msgs.cmf

```diff
@@ -116,3 +116,55 @@ Msg ImmutableDbValue 4002 {
 Msg ImmutableDbVersion 4003 {
     uint64 block_id
 }
+
+# Merkle Tree Data
+
+Msg MerkleBlockValue 5000 {
+    fixedlist uint8 32 root_hash
+    list KeyHash hashed_added_keys
+    list KeyHash hashed_deleted_keys
+}
+
+Msg NibblePath 5001 {
+    uint8 length
+    list uint8 data
+}
+
+Msg BatchedInternalNodeKey 5002 {
+    # The version of this key is the *tree* version, not block version
+    uint64 version
+    NibblePath path
+}
+
+Msg LeafChild 5003 {
+    fixedlist uint8 32 hash
+    VersionedKey key
+}
+
+Msg InternalChild 5004 {
+    fixedlist uint8 32 hash
+    uint64 version
+}
+
+Msg BatchedInternalNodeChild 5005 {
+    oneof {
+        LeafChild
+        InternalChild
+    } child
+}
+
+Msg BatchedInternalNode 5006 {
+    # Contains a 1 where a child node is present, 0 otherwise
+    uint32 bitmask
+    list BatchedInternalNodeChild children
+}
+
+# An index to the latest version of a key.
+#
+# A sentinel of 0 is used to mark the key as deleted. We don't need tombstones as we never rewrite
+# the same key after it has been deleted. The use of a sentinel is to optimize the size and
+# serialization performance of this index.
```

Comment:
Does that imply that once a key is deleted, the user can no longer set a new value for it in a future block?

Reply:
I actually don't believe that comment is correct. I don't see any problem with rewriting a key after its been deleted. I will change it.